src/index.ts

```diff
@@ -1569,16 +1579,26 @@ export class InfluxDB {
    * Creates options to be passed into the pool to query databases.
    * @private
    */
-  private _getQueryOpts(params: any, method = "GET"): any {
-    return {
-      method,
-      path: "/query",
-      query: {
-        p: this._options.password,
-        u: this._options.username,
-        ...params,
-      },
+  private _getQueryOpts(params: any, method = "GET", partOfBody = false): any {
+    const query = {
+      u: this._options.username,
+      p: this._options.password,
+      ...params,
     };
+
+    if (method === "POST" && partOfBody) {
+      return {
+        method,
+        path: "/query",
+        body: querystring.stringify(query),
```

Comment:
The username and password is still required to be passed in the query rather than in the body so need to be persisted.

From InfluxDB docs example:

```
curl -XPOST 'http://localhost:8086/query?u=myusername&p=notmypassword' --data-urlencode 'q=CREATE DATABASE "mydb"'
```

Adding the following sorts it:

```diff
  path: "/query",
  body: querystring.stringify(query),
+ query: {u: this._options.username, p: this._options.password}
```

Reply:
PS. the u & p could then also be dropped from sending as part of the body

Reply:
Yeah makes sense. Reading the docs I assumed you can pass everything down as part of the body. I will adjust it later today. Appreciate the review!

Reply:
@bencevans pushed a fix. Hopefully that sorts it out.